docs/tutorial/flask.rst

```diff
@@ -0,0 +1,530 @@
+make use of the new code:
+
+.. literalinclude:: code/flask/task.yaml
+    :language: bash
+    :start-after: [docs:refresh-deployment]
+    :end-before: [docs:refresh-deployment-end]
+    :dedent: 2
+
+Wait for ``juju status`` to show that the App is ``active`` again. Verify that
+the new configuration has been added using
+``juju config flask-hello-world | grep -A 6 greeting:`` which should show
+the configuration option.
+
+.. note::
+
+    The ``grep`` command extracts a portion of the configuration to make
+    it easier to check whether the configuration option has been added.
+
+Using ``curl http://flask-hello-world --resolve flask-hello-world:80:127.0.0.1``
+shows that the response is still ``Hello, world!`` as expected.
+The greeting can be changed using Juju:
+
+.. literalinclude:: code/flask/task.yaml
+    :language: bash
+    :start-after: [docs:change-config]
+    :end-before: [docs:change-config-end]
+    :dedent: 2
+
+``curl http://flask-hello-world --resolve flask-hello-world:80:127.0.0.1``
+now returns the updated ``Hi!`` greeting.
+
+.. note::
+
+    It might take a short time for the configuration to take effect.
+
+Integrate with a database
+=========================
+
+Now let's keep track of how many visitors your application has received.
+This will require integration with a database to keep the visitor count.
+This will require a few changes:
+
+* We will need to create a database migration that creates the ``visitors`` table
+* We will need to keep track how many times the root endpoint has been called
+  in the database
+* We will need to add a new endpoint to retrieve the number of visitors from the
+  database
+
+Let's start with the database migration to create the required tables.
+The charm created by the ``flask-framework`` extension will execute the
+``migrate.py`` script if it exists. This script should ensure that the
+database is initialised and ready to be used by the application. We will
+create a ``migrate.py`` file containing this logic.
+
+Go back out to the tutorial root directory using ``cd ..``, open the ``migrate.py``
+file using a text editor and paste the following code into it:
+
+.. literalinclude:: code/flask/visitors_migrate.py
+    :language: python
+
+.. note::
+
+    The charm will pass the Database connection string in the
+    ``POSTGRESQL_DB_CONNECT_STRING`` environment variable once
+    postgres has been integrated with the charm.
+
+Open the ``rockcraft.yaml`` file in a text editor and update the version to ``0.3``.
+
+To be able to connect to postgresql from the Flask app the ``psycopg2-binary``
+dependency needs to be added in ``requirements.txt``. The app code also needs
+to be updated to keep track of the number of visitors and to include a new
+endpoint to retrieve the number of visitors to the app. Open ``app.py`` in
+a text editor and replace its contents with the following code:
+
+.. literalinclude:: code/flask/visitors_app.py
```

Comment:
On my screen this block breaks the full-page height guideline for code snippets. What do you think about wrapping it in a `.. collapse` so it's hidden by default?

Reply:
That's fine with me, pinging @jdkandersson in case he has any concerns or objections